### PR TITLE
feat: python lint and format action with black and ruff

### PR DESCRIPTION
**Pull request by Nemental**

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+name: Lint
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
```
Comment (xFuture603) [inline .github/workflows/lint.yml:7]: Could you add `workflow_dispatch:` here too?

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+      - name: Get changed files
+        id: changed-files
+        uses: tj-actions/changed-files@v44
+        with:
+          files: "**/*.py"
+
+      - name: Black formatter
+        if: ${{ steps.changed-files.outputs.all_changed_files != '' }}
+        uses: psf/black@stable
```
Comment (xFuture603) [inline .github/workflows/lint.yml:23]: Could you pin the version to the latest version so that Renovate can pick up these versions and update them?

**Review by xFuture603 — CHANGES_REQUESTED** [review]
Thank you for your pull request! I've added two things I would like to see. Otherwise I've tested the code and it looks good to me.

**Review by Nemental — COMMENTED** [review]
Resolved your suggestions @xFuture603

**Pull request merged**